src/target/stm32f4.c

```diff
@@ -242,6 +245,47 @@ bool stm32f4_probe(target *t)
 	return false;
 }
 
+static bool gd32f4_attach(target *t)
```

Comment:
Given this is a fixed map, rather than creating churn on attach it might be better to construct the map once during probe and let attach be handled normally by `cortexm_attach()` directly. This isn't a blocker if you think keeping the map building code in its own attach would be better somehow.